### PR TITLE
Api provides visitors functionality to display local (Backyard conspiracies) articles

### PR DESCRIPTION
**Pull request by arunbhalli**
PT:  https://www.pivotaltracker.com/story/show/178176049
### User Story:
```
As an API
In order to give visitors access to Backyard conspiracy articles
I need a restricted endpoint for all Backyard articles
```
____________
### Changes Proposed
- Open up ROUTE for  `api/backyards`
- Index action backyard articles with location filtration
- Adds backyard, theme, and location attributes
- Adds conditional rendering and validation based on backyard value
_____________
### Gems/ dependacies 
-  Geocoder gem

### What we Learned
- how to use conditional validation for backyard article
- using scopes
- testing/mocking geocoder response

### CODE REVIEW

File: app/models/article.rb [hunk 1]
```diff
@@ -1,9 +1,22 @@
 class Article < ApplicationRecord
-  validates_presence_of :title, :teaser, :body, :category
-  validates_inclusion_of :premium, in: [false, true]
-  scope :most_recent, -> { order(updated_at: :desc) }
+  validates_inclusion_of :backyard, in: [false, true]
```
Comment (DavDan1) [inline app/models/article.rb:2]: How does this work?

File: app/models/rating.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 class Rating < ApplicationRecord
   validates_presence_of :rating, :article_id
-  belongs_to :article
+  belongs_to :article, -> { where backyard: false }
```
Comment (DavDan1) [inline app/models/rating.rb:3]: Can you explain?

File: spec/requests/api/visitor/can_respond_with_list_of_backyard_articles_spec.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  end
+ 
+
+  describe 'Successfully' do
+    let!(:backyard_articles) { 3.times { create(:backyard_article) } }
+    before do
+      stub_request(:get, 'https://nominatim.openstreetmap.org/reverse?accept-language=en&addressdetails=1&format=json&lat=59.32021&lon=18.37827')
+      .with(
+        headers: {
+          'Accept' => '*/*',
```
Comment (DavDan1) [inline spec/requests/api/visitor/can_respond_with_list_of_backyard_articles_spec.rb:14]: what does the * do?

Comment (LucasKnudsen) [reply spec/requests/api/visitor/can_respond_with_list_of_backyard_articles_spec.rb:14]: It's a wildcard, accepting everything

File: app/models/rating.rb (hunk 2 above)
Comment (LucasKnudsen) [reply app/models/rating.rb:3]: It's scoping down the association to a certain kind of articles that Rating belong to

File: app/models/article.rb (hunk 1 above)
Comment (LucasKnudsen) [reply app/models/article.rb:2]: Validates that the backyard attribute is either true or false

File: app/models/rating.rb (hunk 2 above)
Comment (DavDan1) [reply app/models/rating.rb:3]: nice! thanks

File: spec/requests/api/visitor/can_respond_with_list_of_backyard_articles_spec.rb (hunk 3 above)
Comment (DavDan1) [reply spec/requests/api/visitor/can_respond_with_list_of_backyard_articles_spec.rb:14]: sweet! thanks

File: app/models/article.rb [hunk 4]
```diff
@@ -1,9 +1,22 @@
...
+
+  # Normal articles
+  validates :category, :body, :teaser, presence: true, unless: :is_backyard?
+  validates :premium, inclusion: { in: [false, true] }, unless: :is_backyard?
+  validates :category, inclusion: { in: %w[Science Aliens Covid Illuminati Politics Hollywood] }, unless: :is_backyard?
   has_one_attached :image
+  has_many :ratings
+
+  # Backyard articles
+  validates :theme, :location, presence: true, if: :is_backyard?
```
Comment (offcast-dev) [inline app/models/article.rb:17]: can you just say backyard instead of is_backyard?

File: db/seeds.rb [hunk 5]
```diff
@@ -44,20 +44,32 @@
 
 categories = %w[Science Aliens Covid Illuminati Politics Hollywood]
 
+themes = %w[Science Aliens Covid Trump Cats CIA Singles]
+
+puts 'Creating journalist...'
```
Comment (offcast-dev) [inline db/seeds.rb:49]: cool idea

**Review by offcast-dev — APPROVED** [review]
Small question

File: app/models/article.rb (hunk 1 above)
Comment (lukeperrin10) [reply app/models/article.rb:2]: It checks whether the attribute for backyard is true or false, this works kind of in the same way we have with the premium attribute

File: app/models/rating.rb (hunk 2 above)
Comment (lukeperrin10) [reply app/models/rating.rb:3]: Here this stops backyard articles having ratings.

File: spec/requests/api/visitor/can_respond_with_list_of_backyard_articles_spec.rb (hunk 3 above)
Comment (lukeperrin10) [reply spec/requests/api/visitor/can_respond_with_list_of_backyard_articles_spec.rb:14]: No idea, this is part of the headers that we add to stub out the calls. It's from Webmock.

File: app/models/article.rb (hunk 4 above)
Comment (LucasKnudsen) [reply app/models/article.rb:17]: Wasn't that simple :( Potentially with a lambda or shit

File: app/models/article.rb (hunk 1 above)
Comment (DavDan1) [reply app/models/article.rb:2]: Thanks

**Review by 1CIM — APPROVED** [review]

**Pull request merged**